### PR TITLE
Adds Hardsuit Specific Flag for Medical Sprays / Patches

### PR DESCRIPTION
**Pull request by JamieD1**
Adds Hardsuit Specific Flag for Medical Sprays / Patches

#### Changelog

:cl:  
tweak: Adds Hardsuit Specific Flag for Medical Sprays / Patches
/:cl:

### CODE REVIEW

File: code/__DEFINES/obj_flags.dm [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #define IN_STORAGE				(1<<11) //is this item in the storage item, such as backpack? used for tooltips
 #define SURGICAL_TOOL			(1<<12)	//Tool commonly used for surgery: won't attack targets in an active surgical operation on help intent (in case of mistakes)
 #define UNCATCHABLE				(1<<13)
+#define MEDRESIST				(1<<14)
```
Comment (jcatjmeow) [inline code/__DEFINES/obj_flags.dm]:
```suggestion
#define MEDRESIST				(1<<14) //This item will block medical sprays when worn
```

**Review by jcatjmeow — CHANGES_REQUESTED** [review]
wouldn't hurt to comment on uncatchable too although that's decently self explanatory

File: code/modules/reagents/reagent_containers/patch.dm [hunk 2]
```diff
@@ -13,6 +13,10 @@
 
 /obj/item/reagent_containers/pill/patch/attack(mob/living/L, mob/user)
 	if(ishuman(L))
+		var/mob/living/carbon/human/H = L
+		if(H.wear_suit?.item_flags & MEDRESIST)
+			to_chat(H, "<span class='warning'>[src] cannot be applied through a hardsuit!</span>")
```
Comment (Amelia0010) [inline code/modules/reagents/reagent_containers/patch.dm]:
```suggestion
			to_chat(H, "<span class='warning'>[src] cannot be applied through [H.wear_suit]!</span>")
```

File: code/modules/reagents/reagent_containers/medspray.dm [hunk 3]
```diff
@@ -37,6 +37,11 @@
 		to_chat(user, "<span class='warning'>[src] is empty!</span>")
 		return
 
+	var/mob/living/carbon/human/H = user
+	if(H.wear_suit?.item_flags & MEDRESIST)
+		to_chat(H, "<span class='warning'>[src] cannot be applied through a hardsuit!</span>")
```
Comment (Amelia0010) [inline code/modules/reagents/reagent_containers/medspray.dm]:
```suggestion
		to_chat(H, "<span class='warning'>[src] cannot be applied through [H.wear_suit]!</span>")
```

**Review by Amelia0010 — CHANGES_REQUESTED** [review]

**Comment by Reed0506** [comment]
wow, code that isnt shittily made and doesnt cause problems unrelated to the original thing, incredible!

**Pull request merged**